src/Widgets/EndSessionDialog.vala

```diff
@@ -119,17 +116,44 @@ public class Session.Widgets.EndSessionDialog : Gtk.Dialog {
                 server.closed ();
                 destroy ();
             });
+
+            action_area.add (confirm_restart);
         }
 
-        var cancel = add_button (_("Cancel"), Gtk.ResponseType.CANCEL) as Gtk.Button;
+        action_area.add (cancel);
+        action_area.add (confirm);
+
+        var grid = new Gtk.Grid ();
+        grid.column_spacing = 12;
+        grid.margin = 12;
+        grid.margin_top = 6;
+        grid.attach (image, 0, 0, 1, 2);
+        grid.attach (primary_label, 1, 0);
+        grid.attach (secondary_label, 1, 1);
+        grid.attach (action_area, 0, 2, 2);
+
+        var titlebar = new Gtk.HeaderBar ();
+        titlebar.custom_title = new Gtk.Grid ();
+        titlebar.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        deletable = false;
+        resizable = false;
+        skip_taskbar_hint = true;
+        skip_pager_hint = true;
+        type_hint = Gdk.WindowTypeHint.DIALOG;
+        get_style_context ().add_class ("rounded");
+        set_keep_above (true);
+        set_position (Gtk.WindowPosition.CENTER_ALWAYS);
```

Comment:
Don't we just want `CENTER`? From [Valadoc](https://valadoc.org/gtk+-3.0/Gtk.WindowPosition.html):

>Note that using `CENTER_ALWAYS` is almost always a bad idea. It won’t necessarily work well with all window managers or on all windowing systems.

```suggestion
        set_position (Gtk.WindowPosition.CENTER);
```

Reply:
It looks like we just use `CENTER` in the polkit dialog at least: https://github.com/elementary/pantheon-agent-polkit/blob/b44ccc32f47570d49167aef41072ead453a2449f/src/PolkitDialog.vala#L54